tests/avocado/avocado_qemu/base.py

```diff
@@ -0,0 +1,60 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (C) 2017 Red Hat Inc.
+#
+# Authors:
+#  Amador Pahim <apahim@redhat.com>
+
+"""
+Base class to provide Avocado tests a simple VM.
+"""
+
+import logging
+import os
+import sys
+import uuid
+
+
+from avocado import Test
+from avocado.utils import process
+from avocado.utils import path as utils_path
+
+sys.path.append(os.path.join(os.path.dirname(__file__), '..', '..', '..',
+                'scripts'))
+import qemu
+
+
+def _get_qemu_bin(arch):
```

Comment:
As we've talked before, different qemu testing tools do this differently.  I think we have the opportunity to do it for once and make it a standard.  By writing a module that understands the build configuration, we can even add some command line capabilities, in such that it could be called like:

```
python -m qemu.buildconf list-build-targets-bin
```

Or

```
python -m qemu.buildconf default-target-bin
```

Which could be used in the other scripts.